### PR TITLE
Add dependencies via reflection

### PR DESCRIPTION
**Pull request by gordon-klotho**
This should make it much easier to add dependencies in many (most?) cases.

### Standard checks

- **Unit tests**: Any special considerations?
- **Docs**: Do we need to update any docs, internal or public?
- **Backwards compatibility**: Will this break existing apps? If so, what would be the extra work required to keep them working?

### CODE REVIEW

File: pkg/core/resource_graph.go [hunk 1]
```diff
@@ -10,58 +12,120 @@ type (
...
+//
+// Supported field types (`*T` is a struct that implements Resource)
+// - `SingleDependency   Resource`
+// - `SpecificDependency *T`
+// - `DependencyArray  []Resource`
+// - `SpecificDepArray []*T`
+// - `DependencyMap  map[string]Resource`
+// - `SpecificDepMap map[string]*T`
+func (rg *ResourceGraph) AddDependenciesReflect(source Resource) {
+	rv := reflect.ValueOf(source)
```
Comment (yuval-klotho) [inline pkg/core/resource_graph.go:91]: I find these hypershort names hard to follow through the method. I'd prefer `resVal` etc.

**Comment by yuval-klotho** [comment]
this LGTM, but I'd like to get input from Jordan and David as well to make sure we all have consensus around it

**Comment by jhsinger-klotho** [comment]
yeah im fine going this route, theres still scenarios in which we would need to invoke AddDependency though i believe, but well discover those

**Review by DavidSeptimus-Klotho — APPROVED** [review]
It seems neat to me. I'm curious to see how the magical aspect of this plays out over time (vs explicit dependencies).
